File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1]*(len(nums))
        for i in range(1,len(nums)):
            res[i] = res[i-1] * nums[i-1]
        postFix = 1
        for j in range(len(nums)-1 , -1 ,-1):
            res[j]*=postFix
            postFix *= nums[j]

        return res

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int,int>mp ;
        for(auto it : nums){
            mp[it]++;
        }
        for(auto i : mp){
            if (i.second > 1){
                return true;
            }
        }
        return false;
    }
};